### PR TITLE
prettier integration

### LINKED ISSUE #379: Prettier integration
#### My reason:

A few days ago, when I was working on an issue #327, My IDE changed the indentation of the code and made hard to understand. Is it useful add Prettier (https://prettier.io/) to maintain code presentation?

### PR DESCRIPTION
**Pull request by ayusharma**
**Type:** build

The following has been addressed in the PR:
- `npm run prettier-list`  list file different from prettier config.
- `npm run prettier-write` write the prettier changes.

**Description:**

Resolves #379

Should I add prettier on precommit hook ?

### CODE REVIEW

**Comment by ayusharma** [comment]
CI is failing due to `prettier-eslint`.

**Comment by juanpicado** [comment]
I'll try to check as soon as possible, the internet where I'm now is quite bad.

File: package.json [hunk 1]
```diff
@@ -135,7 +138,9 @@
     "dev:webui": "babel-node tools/dev.server.js",
     "build:webui": "npm run pre:webpack && webpack --config tools/webpack.prod.config.babel.js",
     "build:docker": "docker build -t verdaccio . --no-cache",
-    "build:docker:rpi": "docker build -f Dockerfile.rpi -t verdaccio:rpi ."
+    "build:docker:rpi": "docker build -f Dockerfile.rpi -t verdaccio:rpi .",
+    "prettier-list":"./node_modules/.bin/prettier -l --config .prettierrc.json './{test,src,tools}/**/*.js'",
```
Comment (juanpicado) [inline package.json]: `./node_modules/.bin/` is not need it. `npm` map `prettier` from that location anyway

**Comment by juanpicado** [comment]
It fails due to the massive changes in the code base. I'd mind if you do it, but, in any case should be from `3.x` branch because rebase from master is gonna be painful.

**Comment by juanpicado** [comment]
git hook? With this https://www.npmjs.com/package/husky ?

**Comment by ayusharma** [comment]
Yes. Using husky. https://prettier.io/docs/en/usage.html#pre-commit-hook

File: package.json (hunk 1 above)
Comment (ayusharma) [reply package.json]: Okay. I'll update this.

**Comment by juanpicado** [comment]
Cool do it on v3.x then :)

**Comment by ayusharma** [comment]
I'll do it.

**Comment by juanpicado** [comment]
Perfect 👍  I'm ready for a super mega Pull Request 🌪

**Comment by ayusharma** [comment]
@juanpicado  I am facing some issue on running tests on `v3.x` 

```
  1) functional test verdaccio "before all" hook:
     Error: connect ECONNREFUSED 127.0.0.1:55551
      at Object.exports._errnoException (util.js:1022:11)
      at exports._exceptionWithHostPort (util.js:1045:20)
      at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1087:14)
```

Latest commit in 3.x is
```
commit f8381ba2dfb0ed7865727505cc0827d1a1ac528e (HEAD, upstream/3.x, issue-379-copy)
Merge: 3a718b9 0358fc9
Author: Juan Picado @jotadeveloper <juanpicado19@gmail.com>
Date:   Sun Oct 22 12:46:17 2017 +0200

    Merge pull request #291 from verdaccio/local-storage-as-dependency

    Local storage as dependency plugin

```

Please take a look.

**Comment by juanpicado** [comment]
Before proceed perhaps you can join the slack channel to discuss more in detail https://verdaccio-npm.slack.com

**Comment by ayusharma** [comment]
Is this invite only? Send me an invite at ayush.aceit@gmail.com

**Comment by Meeeeow** [comment]
I don't recommend add such tool, it runs like a black box.

With the experience in my company, prettier auto format will broken alot of things if we use different editor or IDE.

And there is no way to import all prettier rule to all IDE, so, auto format before commit, format back or got warning in the next time to edit... Really annoying...

For example, with some editor's plugin, prettier will auto format package.json, then change back while another one add packages to package.json.

And it will destroy some nice feature, prettier with file watcher often make a lot of external chane in WebStorm built in file history.

Also, if somebody use editor such as notepad++ or vim, it will cause more problem. editor will say file changed, reload or not, if you have unsaved changes, thats's really bad.
Nobody want read it's pretty long document and remember rules or set them into editor by hand.

So i recommend use ESLint to do pre-commit check, but never auto change ANYTHING. Because not all people like it's defaule rule, don't just follow without ask why.

**Comment by juanpicado** [comment]
I think @Meeeeow is right with that. As I said before I never use it. @ayusharma I added you to the slack channel

**Pull request closed without merging**